compiler/compiler.go

```diff
@@ -546,9 +546,14 @@ func compile(n semantic.Node, subst map[uint64]semantic.MonoType, scope Scope) (
 		if err != nil {
 			return nil, err
 		}
-		return &interpolatedEvaluator{
+		val := &interpolatedEvaluator{
```

Comment:
Is `val` an `evaluator` or a `value`? Is there a better, more detailed name we could give this variable?